### PR TITLE
Avoid using OBJ_WB_UNPROTECT

### PR DESCRIPTION
**Pull request by casperisfine**
Not that it makes a big difference here, but it's just unnecessary.

cc @peterzhu2118

### CODE REVIEW

File: struct.c [hunk 1]
```diff
@@ -413,18 +413,16 @@ struct_make_members_list(va_list ar)
 {
     char *mem;
     VALUE ary, list = rb_ident_hash_new();
-    st_table *tbl = RHASH_TBL_RAW(list);
 
-    RBASIC_CLEAR_CLASS(list);
```
Comment (XrXr) [inline struct.c:418]: I think you want to keep the CLEAR_CLASS to hide the object from ObjectSpace.each_object

Comment (casperisfine) [reply struct.c:418]: We were actually questioning the usefulness of it. It's frozen, is it a big deal if you can find it though ObjectSpace?

Comment (casperisfine) [reply struct.c:418]: Sorry, this one isn't frozen, but ephemeral, we don't keep a reference on it.

Comment (XrXr) [reply struct.c:418]: I like to be conservative with this stuff. Not necessarily against removing it, but please at least do it in a separate commit. This is a separate concern from OBJ_WB_UNPROTECT.

Comment (casperisfine) [reply struct.c:418]: Ok, added back.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**